test/functional/test_framework/test_framework.py

```diff
@@ -672,6 +673,22 @@ def generatetodescriptor(self, generator, *args, sync_fun=None, **kwargs):
         sync_fun() if sync_fun else self.sync_all()
         return blocks
 
+    def create_outpoints(self, node, *, outputs):
```

Comment:
Review ACK 7154542

The proposed refactor is a commendable improvement, streamlining test cases by leveraging the ```create_outpoints(...)``` function to consolidate calls to ```nodes.sendtoaddress()``` and ```find_vout_for_address()``` into a single invocation.

Suggestion: It might be beneficial to add documentation clarifying that the ```outputs``` argument should be an array of dictionaries following the format ```{address: amount_to_send}```. This would enhance usability for future users, eliminating the need to inspect example invocations.

Reply:
> Suggestion: It might be beneficial to add documentation clarifying that the outputs argument should be an array of dictionaries following the format {address: amount_to_send}. This would enhance usability for future users, eliminating the need to inspect example invocations.

Good idea. Made the following (simple) change, I hope that's clear enough for users:
```diff
diff --git a/test/functional/test_framework/test_framework.py b/test/functional/test_framework/test_framework.py
index 045fd2f5cb..70b3943478 100755
--- a/test/functional/test_framework/test_framework.py
+++ b/test/functional/test_framework/test_framework.py
@@ -699,7 +699,7 @@ class BitcoinTestFramework(metaclass=BitcoinTestMetaClass):
         return blocks

     def create_outpoints(self, node, *, outputs):
-        """Send funds to a given list of address/amount targets using the bitcoind
+        """Send funds to a given list of `{address: amount}` targets using the bitcoind
         wallet and return the corresponding outpoints as a list of dictionaries
         `[{"txid": txid, "vout": vout1}, {"txid": txid, "vout": vout2}, ...]`.
         The result can be used to specify inputs for RPCs like `createrawtransaction`,

```